torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -538,3 +539,40 @@ def wrapper(*args, **kwargs):
 
 # Import all builtin sharded ops
 from .ops import *  # noqa: F403
+
+def _reshard_output(
+        module: torch.nn.Module,
+        resharding_spec: ShardingSpec) -> torch.nn.Module:
+    """
+    Hook a module with local shards collection in the forward pass according
+    to the given ``resharding_spec``.
+
+    Args:
+        module (:class:`torch.nn.Module`): Module whose output needs to be resharded.
+        resharding_spec (:class:`torch.distributed._sharding_spec.ShardingSpec`): The
+            specification describing how the sharded tensor will be resharded.
+
+    Returns:
+        A :class:`torch.nn.Module` object with collection API hooked.
+    """
+    def hook_func(_module, _input, output):
+        return output.reshard(resharding_spec)
+    module.register_forward_hook(hook_func)
+    return module
+
+
+def _collect_local_shard(module: torch.nn.Module) -> torch.nn.Module:
```

Comment:
We need tests for these two new APIs too.

Reply:
Do we feel the test in the sharded_linear and megatron prototype is enough?

Reply:
Update: After a second thought, I think it still makes sense to create a dedicated unit test for them. I will update the unit test plan.